### PR TITLE
Updated CIFAR-10 ResNet example to use BatchNorm instead of LayerNorm

### PR DESCRIPTION
**Pull request by menzHSE**
This PR updates the CIFAR-10 ResNet example to use BatchNorm instead of LayerNorm. It is tested with mlx@026ef9a but should probably wait until https://github.com/ml-explore/mlx/pull/385 (remove `retain_graph` flag) appears in an official release. I am assuming, this will be in mlx 0.0.8. 

**Changes**
* Replaced ```nn.LayerNorm``` by ```nn.BatchNorm``` (test accuracy 0.807 after 100 epochs -> 0.833 after 30 epochs)
* Decreased default number of epochs from 100 to 30 (no improvement observed after approx. 25 epochs)
* Updated code comments and ```README.md```
* Updated ```requirements.txt``` to require ```mlx>=0.0.8```

**TODO**

- [x] Test with next official mlx release and verify ```mlx>=0.0.8``` in ```requirements.txt``` is still valid

### CODE REVIEW

**Comment by awni** [comment]
@menzHSE did you record the "Previous" results on the same machine? In general it's expected that there can be considerable variability between machines.

**Comment by menzHSE** [comment]
@awni I did not record the "Previous" results. It is taken from the original ```README.md```. I have tried to reproduce it with earlier mlx versions (up to 0.0.2) but am always getting the same throughput. So, whoever reported the initial results must have a better M1 MBPro than mine ;-)

**Comment by awni** [comment]
Well that is very strange... let me time it on my machine..

**Comment by awni** [comment]
I am on main right now with a 32GB M1 Max. Clearly that seems to help a lot...  I don't have access to a 16GB M1 pro, but I will also plan try on an M1 Mini and/or M2 Mini

```
Epoch 00 [020] | Train loss 2.215 | Train acc 0.223 | Throughput: 624.82 images/second
```

**Review by awni — APPROVED** [review]
Looks great, thanks for updating this.

I will wait until 0.0.8 to merge it.

Also the perf for this example needs a lot of work 😞 , even the 400 im/s was really bad...

**Comment by menzHSE** [comment]
Tested with mlx 0.0.9. Updated requirements.txt to require mlx>=0.0.9. Rebased and force-pushed. I think it is ready to be merged.

**Comment by menzHSE** [comment]
Tested with mlx 0.0.9. Ready to be merged.

**Comment by awni** [comment]
Thanks @menzHSE !
